core/hoverfly_funcs.go

```diff
@@ -117,6 +114,65 @@ func (hf *Hoverfly) GetResponse(requestDetails models.RequestDetails) (*models.R
 	return &response, nil
 }
 
+func (hf *Hoverfly) applyBodyTemplating(requestDetails *models.RequestDetails, response *models.ResponseDetails, cachedResponse *models.CachedResponse) (string, error) {
+	var template *raymond.Template
+	if cachedResponse != nil && cachedResponse.ResponseTemplate != nil {
+		template = cachedResponse.ResponseTemplate
+	} else {
+		// Parse and cache the template
+		template, _ = hf.templator.ParseTemplate(response.Body)
+		if cachedResponse != nil {
+			cachedResponse.ResponseTemplate = template
+		}
+	}
+
+	return hf.templator.RenderTemplate(template, requestDetails, hf.state.State)
+}
+
+func (hf *Hoverfly) applyHeadersTemplating(requestDetails *models.RequestDetails, response *models.ResponseDetails, cachedResponse *models.CachedResponse) (map[string][]string, error) {
+	var headersTemplates map[string][]*raymond.Template
+	if cachedResponse != nil && cachedResponse.ResponseHeadersTemplates != nil {
+		headersTemplates = cachedResponse.ResponseHeadersTemplates
+	} else {
+		var header []*raymond.Template
```

Comment:
For brevity, can you remove this line and use short variable declaration like: `header := make([]*raymond.Template, len(v))`?

Reply:
It good proposal, but `response.Headers` below has signature `map[string][]string`. Every header has different length slice of strings.